src/anndata/_core/anndata.py

```diff
@@ -541,7 +542,17 @@ def shape(self) -> tuple[int, int]:
         return self.n_obs, self.n_vars
 
     @property
-    def X(self) -> np.ndarray | sparse.spmatrix | SpArray | ArrayView | None:
+    def X(
+        self,
+    ) -> (
+        CSRDataset
+        | CSCDataset
+        | np.ndarray
+        | sparse.spmatrix
```

Comment:
```suggestion
        | sparse.csr_matrix
        | sparse.csc_matrix
```

Reply:
So somewhere, we are sticking a `dia_matrix` into an `AnnData` object.  For now I'm going to leave this, and file an issue.

Reply:
Nope! too quick! this caught a bug in a test! good!